### PR TITLE
fix: fix default time formatting issue

### PR DESCRIPTION
**Pull request by butcherZ**
### What does it do?

add `handleTimeChange` function to format time when needed.

### Why is it needed?

Input (Display) Format:

The`TimePicker` only supports the time format "HH:MM" (hours and minutes).
The default value set in the Content-Type Builder (CTB) comes in the format `HH:mm:ss.SSS`
To display the default value correctly, we need to convert it from `HH:mm:ss.SSS` to `HH:MM`.

Output (Submission) Format:

When a user selects or changes a time in the `TimePicker`, it provides the time in `HH:MM` format.
However, the backend expects to receive the time in the full `HH:mm:ss.SSS` format.
Therefore, we need to convert the user-selected time from `HH:MM` back to `HH:mm:ss.SSS` before sending it to the backend.

In summary:

For display: Convert `HH:mm:ss.SSS` → `HH:MM`
For submission: Convert `HH:MM` → `HH:mm:ss.SSS`

### How to test it?

Manually: 
- Create a `Date` field with type `time` in any collection in the Content type builder
- Go to advanced settings and select a `default value`
- Save
- Go to Content Manager, create a new entry with the default value
- There should be no errors

Unit test:

`cd strapi/packages/core/content-type-builder`
run `yarn test:front `

### Related issue(s)/PR(s)

#20875 

DX-1592

### CODE REVIEW

File: packages/core/content-type-builder/admin/src/utils/timeFormat.ts
Comment (innerdvations) [inline packages/core/content-type-builder/admin/src/utils/timeFormat.ts:13]:
```suggestion
// The backend sends a value which has the following format: '00:45:00.000'
// but the time picker only supports hours & minutes so we need to mutate the value
```

**Review by innerdvations — APPROVED** [review]
Looks good to me, tested and works! Wait for a front-end review before merging though just in case I'm missing something :)

File: packages/core/content-type-builder/admin/src/utils/timeFormat.ts
Comment (Bassel17) [inline packages/core/content-type-builder/admin/src/utils/timeFormat.ts:6]:
```suggestion
  value?: string;
```
you can do this instead just a little neater

File: packages/core/content-type-builder/admin/src/utils/timeFormat.ts
Comment (Bassel17) [inline packages/core/content-type-builder/admin/src/utils/timeFormat.ts:25]: you can choose to do this  you don't have to explicitly return undefined
```suggestion
  if (!value) return;
```
or just use 

```
if (value) {
  return value.split(':').length > 2 ? removeSeconds(value) : value;
}
```

File: packages/core/content-type-builder/admin/src/utils/timeFormat.ts
Comment (Bassel17) [inline packages/core/content-type-builder/admin/src/utils/timeFormat.ts:24]:
```suggestion
const formatTimeForInput = (value?: string): string | undefined => {
```

File: packages/core/content-type-builder/admin/src/utils/timeFormat.ts
Comment (Bassel17) [inline packages/core/content-type-builder/admin/src/utils/timeFormat.ts:29]: Same ideas for this one as `formatTimeForInput`

File: packages/core/content-type-builder/admin/src/utils/timeFormat.ts
Comment (Bassel17) [inline packages/core/content-type-builder/admin/src/utils/timeFormat.ts:41]:
```suggestion
  time?: string,
```

**Review by Bassel17 — CHANGES_REQUESTED** [review]
Looks great! and seems to be working 
just left a few small comments

**Review by Bassel17 — APPROVED** [review]
LGTM!

there seems to be a failing e2e test suite, am betting it's a flaky one, you can attempt to re-run it, it will probably pass

**Pull request merged**
